backend/src/tasks/lookingGlass.ts

```diff
@@ -0,0 +1,316 @@
+    if (organizationName === line.name) {
+      const collectionID = line.id;
+      console.log(line.name);
+      // Query LookingGlass for the Threat info
+      const data: LGThreatResponse = await getThreatInfo(collectionID);
+      const responseDomains: Domain[] = await saveAndPullDomains(
+        data,
+        organizationId!,
+        scanId
+      );
+      // Only pull domains that have been seen in the last two months.
+      const twoMonthsAgo = new Date();
+      twoMonthsAgo.setMonth(twoMonthsAgo.getMonth() - 2);
+      console.log(`Pulling domains seen since ${twoMonthsAgo}`);
+      for (const l of data.results) {
+        // Create a dictionary of relevant fields from the API request.
+        const lastSeen = new Date(l.lastSeen);
+        if (lastSeen > twoMonthsAgo) {
+          const val: ParsedLGResponse = {
+            firstSeen: new Date(l.firstSeen),
+            lastSeen: lastSeen,
+            sources: l.sources,
+            ref_type: l.ref.type,
+            ref_right_type: l.ref.right.type,
+            ref_right_id: l.ref.right.id,
+            ref_left_type: l.ref.left.type,
+            ref_left_id: l.ref.left.id,
+
+            right_ticScore: l.right.ticScore,
+            right_classifications: l.right.classifications,
+            right_name: l.right.name,
+
+            left_type: l.left.type,
+            left_ticScore: l.left.ticScore,
+            left_name: l.left.name,
+            vulnOrMal:
+              l.right.classifications[0] === 'Vulnerable Service'
+                ? 'Vulnerability'
+                : 'Malware'
+          };
+
+          // If we've already seen this domain, add this val to the structuredData field of the associated vulnerability
+          if (ipsAndDomains.includes(l.left.name)) {
+            for (const Vuln of vulnerabilities) {
+              //Match current LookingGlass Object with existing Vulnerability
+              if (l.left.name === Vuln.domain.name) {
+                let matches = false;
+                // If the Vuln type already exists for this Domain keep the most recent instance
+                // Loop through the previously created Vulnerabilities structuredData to identify duplicates
+                Vuln.structuredData['lookingGlassData'].forEach(function (
+                  row,
+                  index,
+                  array
+                ) {
+                  // Find a duplicate threat types
+                  if (l.right.name === row.right_name) {
+                    // If the new value is more recent than existing value, update new values firstSeen date from existing values
+                    // first date and then overwrite the existing value
+                    if (lastSeen > row.lastSeen) {
+                      val.firstSeen = row.firstSeen;
+                      array[index] = val;
+                      // if the existing value is more recent than the new value then update the existing values firstSeen from the new value
```

Comment:
I think you should move this comment to within the else statement so it's clear that it applies to that case.
```suggestion
                      // If the existing value is more recent than the new value then update the existing value's firstSeen from the new value
```